python/ray/train/callbacks/logging.py

```diff
@@ -19,6 +20,42 @@
 logger = logging.getLogger(__name__)
 
 
+class PrintCallback(TrainingCallback):
+    """A callback that prints training results to STDOUT.
+
+    Example:
+        >>> # Handle results from two workers.
+        >>> callback = PrintCallback()
+        >>> results = [{"loss": 2.290156}, {"loss": 2.275099}]
+        >>> callback.handle_result(results)
+        {
+            "loss": [
+                2.290156,
+                2.275099
+            ]
+        }
+    """
+
+    def handle_result(self, results: List[Dict], **info):
+        """Prints results to STDOUT.
+
+        Args:
+            results (List[Dict]): List of results from the training
+                function. Each value in the list corresponds to the output of
+                the training function from each worker.
+            **info: kwargs dict for forward compatibility.
+        """
+        # Check that the dictionaries have the same keys.
+        assert all(result.keys() == results[0].keys() for result in results)
+
+        dictionary = collections.defaultdict(list)
+        for result in results:
+            for key in result:
+                dictionary[key].append(result[key])
+
+        print(json.dumps(dictionary, indent=4))
```

Comment:
Let's just keep this simple and print out the raw results for now (similar to the JSON logger callback).

We can add a note in the PR description/elsewhere to keep track of this as a possible enhancement.

Reply:
Addressed